src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.CompletionListUpdater.cs

```diff
@@ -266,6 +266,11 @@ private void AddCompletionItems(List<MatchResult<VSCompletionItem>> list, Cancel
                     MatchResult<VSCompletionItem> bestOrFirstMatchResult;
                     if (filteredItemsBuilder.Count == 0)
                     {
+                        // When we are in suggestion mode and there's nothing in the list matches what user has typed in any ways,
+                        // we should select the SuggestionItem instead.
+                        if (_snapshotData.DisplaySuggestionItem)
+                            return new ItemSelection(SelectedItemIndex: -1, SelectionHint: UpdateSelectionHint.SoftSelected, UniqueItem: null);
```

Comment:
we shou dintroduce a constant for this -1, the repetition is not great.

Reply:
done